File: homework_2.py
try:
    original_cislo = int(input("Zadej přirozené číslo pro převod do dvojkové soustavy: "))
    if original_cislo < 0:
        print("Chyba: Tento algoritmus pracuje pouze s přirozenými čísly a nulou.")
    else:
        binary = ""  # Inicializujeme prázdný řetězec pro binární zápis.
        cislo = original_cislo
        if cislo == 0:
            binary = "0"  # Pokud je vstupní číslo 0, vrátíme "0" jako výsledek.
        else:
            while cislo > 0:
                if cislo % 2 == 0:
                    binary = "0" + binary
                else:
                    binary = "1" + binary
                cislo //= 2

        print(f"Dvojkový zápis čísla {original_cislo} je: {binary}")
except ValueError:
    print("Chyba: Zadali jste neplatný vstup. Zadejte prosím přirozené číslo a nulu.")

File: homework_2.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function convertToBinary(decimalNumber) {

    let number = parseInt(decimalNumber);

    if (isNaN(number) || number < 0) {
        console.error("Chyba: Zadali ste neplatný vstup. Zadajte prosím prirodzené číslo alebo nulu.");
        return
    }

    if (number === 0) {
        return "0"
    }

    let binary = "";

    while (number > 0) {
        binary = (number % 2) + binary;
        number = Math.floor(number / 2);
    }

    return binary
}

rl.question("Zadaj prirodzené číslo alebo nulu pre prevod do dvojkovej sústavy: ", function (input) {
    const output = convertToBinary(input);

    console.log(`Dvojkový zápis čísla ${input} je: ${output}`);
    rl.close();
});
